compiler/src/iree/compiler/Dialect/VM/IR/VMOps.td

```diff
@@ -1196,6 +1197,52 @@ def VM_RodataInlineOp : VM_PureOp<"rodata.inline", [
   ];
 }
 
+def VM_RodataTableOp : VM_PureOp<"rodata.table", [
```

Comment:
we'll likely want to use this for both i32 and i64 indices at some point, so worth adding a discriminator now - a TypeAttr would let us extend the bit widths in the future so we could generate narrower tables because range analysis says we can/etc (`vm.rodata.table i32 : !vm.buffer, !vm.buffer, ...`)

Reply:
Makes sense. Also, figuring I should check, would it be useful to have a non-inline version of this as well? I was initially thinking of trying to let this do both but found it unwieldy to try and put an optional symbol inside the table op.

In which case, would `vm.rodata.table.inline` be a better name?

Reply:
I went ahead and renamed it to `vm.rodata.inline.table` and am going to add another op for the symbol version, as well as a util op for it, the thinking being we can target tables like this at some point before vm by simply packaging up other rodata-like attributes/globals and indexing into them with the table instead of maintaining multiple globals.

Reply:
nit of `vm.rodata.table.inline` (root words, so `vm.rodata.table.inline` is the inline variant of `vm.rodata.table`, etc)